src/miner.cpp

```diff
@@ -71,10 +71,10 @@ bool TxPriorityCompare::operator()(const TxPriority& a, const TxPriority& b)
             const CScCertificate& aCert = dynamic_cast<const CScCertificate&>(*a.get<2>());
             const CScCertificate& bCert = dynamic_cast<const CScCertificate&>(*b.get<2>());
 
-            if (aCert.GetScId() == bCert.GetScId() )
-            {
-                return aCert.quality > bCert.quality;
+            if (a.get<1>() == b.get<1>()) { 
+                return aCert.quality < bCert.quality;
             }
+            return a.get<1>() < b.get<1>();
         } catch (...) {
             LogPrintf("%s():%d - ERROR: cast error\n", __func__, __LINE__ );
             assert("could not cast txbase obj" == 0);
```

Comment:
I know it was already that way before this PR, but the equality check on double type at line 93 (`if (a.get<0>() == b.get<0>())`) is quite bad, would you consider refactoring it?